letta/schemas/tool_rule.py

```diff
@@ -21,6 +21,17 @@ class ChildToolRule(BaseToolRule):
     children: List[str] = Field(..., description="The children tools that can be invoked.")
 
 
+class ConditionalToolRule(BaseToolRule):
+    """
+    A ToolRule that conditionally maps to different child tools based on the output.
+    """
+    type: ToolRuleType = ToolRuleType.conditional
+    default_child: str = Field(..., description="The default child tool to be called")
```

Comment:
I think it's ok for this to be `None` where if it's `None`, then we don't constrain the next tool (it can be any tool)